lib/api/api-request.js

```diff
@@ -189,7 +190,18 @@ function request (opts, callback) {
   }
 
   try {
-    this.dispatch(opts, new RequestHandler(opts, callback))
+    const handler = new RequestHandler(opts, callback)
+
+    // if (opts?.maxRedirections != null && (!Number.isInteger(opts?.maxRedirections) || opts?.maxRedirections < 0)) {
+    //   throw new InvalidArgumentError('maxRedirections must be a positive number')
+    // }
+
+    // if (opts?.maxRedirections > 0) {
+    //   RedirectHandler.buildDispatch(this, opts.maxRedirections)(opts, handler)
+    //   return
+    // }
```

Comment:
Here we need to decide wether we keep this functionality here or completely remove it in pro of the `interceptor`.
If both coexist, they will overlap causing unexpected results, but I do see the benefit of having it built-in.

Anyways, if this is the case, we should document better when and how to use the `interceptor` and I'll need to refactor the tests a little on the `interceptor/redirect`.

On my side, I'm more into removing it but I can see situations when this can be handy.
Any thoughts @ronag @mcollina ?